### PR TITLE
[improve][test] remove WhiteBox on MockZooKeeper

### PR DESCRIPTION
**Pull request by tisonkun**
### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [x] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java [hunk 1]
```diff
@@ -144,10 +144,7 @@ public static MockZooKeeper newInstanceForGlobalZK(ExecutorService executor, int
 
     public static MockZooKeeper newInstance(ExecutorService executor, int readOpDelayMs) {
         try {
-            MockZooKeeper zk = createMockZooKeeperInstance(executor, readOpDelayMs);
-            ObjectInstantiator<ClientCnxn> clientCnxnObjectInstantiator = objenesis.getInstantiatorOf(ClientCnxn.class);
-            Whitebox.setInternalState(zk, "cnxn", clientCnxnObjectInstantiator.newInstance());
-            return zk;
+            return createMockZooKeeperInstance(executor, readOpDelayMs);
```
Comment (tisonkun) [inline testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java:147]: @eolivelli this change was added by you in https://github.com/apache/pulsar/pull/8590. I don't know why we need this `WhiteBox` based mock and after removing it and running several tests, it seems all green.

Could you share some background here?

**Comment by tisonkun** [comment]
cc @Shoothzj @nicoloboschi 

It's interesting that no test failed after remove the lines:

```java
            ObjectInstantiator<ClientCnxn> clientCnxnObjectInstantiator = objenesis.getInstantiatorOf(ClientCnxn.class);
            Whitebox.setInternalState(zk, "cnxn", clientCnxnObjectInstantiator.newInstance());
```

I suppose that since the `MockZooKeeper` already mockup the interface, no caller will access `cnxn` field and thus it's unnecessary. However, it would be nice if the original author @eolivelli can give more background :)

**Review by Shoothzj — APPROVED** [review]

File: testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java (hunk 1 above)
Comment (eolivelli) [reply testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java:147]: Maybe the older versions of Mockito or other mock frameworks needed it

**Review by eolivelli — APPROVED** [review]

**Pull request merged**